app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryCommentsFragment.kt

```diff
@@ -0,0 +1,421 @@
+                            findActivity<MainActivity>().showTip(
+                                if (isUp) (if (0 != result.vote) R.string.vote_up_successfully else R.string.cancel_vote_up_successfully) else if (0 != result.vote) R.string.vote_down_successfully else R.string.cancel_vote_down_successfully,
+                                BaseScene.LENGTH_SHORT,
+                            )
+                            comment.score = result.score
+                            if (isUp) {
+                                comment.voteUpEd = 0 != result.vote
+                                comment.voteDownEd = false
+                            } else {
+                                comment.voteDownEd = 0 != result.vote
+                                comment.voteUpEd = false
+                            }
+                            recomposeScope.invalidate()
+                        }.onFailure {
+                            findActivity<MainActivity>().showTip(R.string.vote_failed, BaseScene.LENGTH_LONG)
+                        }
+                    }
+
+                    suspend fun Context.doCommentAction(comment: GalleryComment) {
+                        val actions = buildAction {
+                            copyComment thenDo { findActivity<MainActivity>().addTextToClipboard(comment.comment.orEmpty().parseAsHtml()) }
+                            if (!comment.uploader && !comment.editable) {
+                                blockCommenter thenDo { showFilterCommenter(comment) }
+                            }
+                            if (comment.voteUpAble) {
+                                (if (comment.voteUpEd) cancelVoteUp else voteUp) thenDo { voteComment(comment, true) }
+                            }
+                            if (comment.voteDownAble) {
+                                (if (comment.voteDownEd) cancelVoteDown else voteDown) thenDo { voteComment(comment, false) }
+                            }
+                            if (!comment.voteState.isNullOrEmpty()) {
+                                checkVoteStatus thenDo { showCommentVoteStatus(comment) }
+                            }
+                        }
+                        dialogState.showSelectItem(*actions.toTypedArray()).invoke()
+                    }
+
+                    GalleryCommentCard(
+                        comment = item,
+                        onUserClick = {
+                            val lub = ListUrlBuilder(
+                                mode = ListUrlBuilder.MODE_UPLOADER,
+                                mKeyword = item.user,
+                            )
+                            navigator.navAnimated(R.id.galleryListScene, lub.toStartArgs(), true)
+                        },
+                        onCardClick = {
+                            coroutineScope.launch {
+                                context.doCommentAction(item)
+                            }
+                        },
```

Comment:
Not working when clicking comment text

Reply:
Yup, comment text has its own OnClickListener
I will try resolve it tomorrow

Reply:
Done